### PR TITLE
feat: implemented Create Owner Component closes #5

### LINKED ISSUE #5: Feat: Create Owner Component
**Description**
Implement the `PetOwner` component for the `PetChain` system. This component will handle all owner-related functionality including registration, profile management, and owner authentication.

**Interface Requirements**
The interface for this component should include `register_pet_owner`, `update_owner_profile`, `is_owner_registered`, `is_pet_owner`.

**Acceptance criteria;**
- Registration of new owners works correctly.
- Profile updates are properly stored.
- Necessary events should be emitted.
- Write a robust test for all functionality.

Approach for solving this will be left to the contributor assigned, please reach out on telegram if you have questions.

### PR DESCRIPTION
**Pull request by aji70**
his PR completes the implementation of the PetOwner component for the PetChain system. The component introduces core functionalities for managing pet owner information (register_pet_owner), updating existing profiles (update_owner_profile), and verifying ownership status (is_owner_registered, is_pet_owner). It defines the PetOwner struct with relevant fields to store owner and pet details, ensuring a scalable and organized data structure for the system. Comprehensive unit tests have been written to validate all functions, covering standard operations and edge cases to guarantee reliability and correctness closes issue #5.

### CODE REVIEW

**Comment by manlikeHB** [comment]
Hi @aji70, You're supposed to create an owner component not a contract, see [this](https://book.cairo-lang.org/ch103-02-00-composability-and-components.html?highlight=component#creating-components).

Create a new folder called `components` and that's where your implementation should go, if you check the `lib.cairo` file, you'll see that the components module has already been declared.

If you have further questions please reach out on telegram.

**Comment by aji70** [comment]
fixed

**Comment by aji70** [comment]
bro please share link to telegram if you are not satisfies with the pr

File: src/components/pet_owner/petowner_component.cairo [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            id == pet_owner.id
+        }
+
+        fn register_pet_owner(
+            ref self: ComponentState<TContractState>,
+            name: ByteArray,
+            email: ByteArray,
+            emergency_contact: ByteArray,
+        ) -> u256 {
+            let caller = get_caller_address();
```
Comment (manlikeHB) [inline src/components/pet_owner/petowner_component.cairo:61]: check caller address not zero

File: src/components/pet_owner/petowner_component.cairo [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                        PetOwnerCreated { owner_address: caller, owner_id: cur_id },
+                    ),
+                );
+
+            cur_id
+        }
+
+        fn update_owner_profile(
+            ref self: ComponentState<TContractState>,
+            id: u256,
```
Comment (manlikeHB) [inline src/components/pet_owner/petowner_component.cairo:93]: no need for this` id`, just the caller address is enough for verification.

File: src/components/pet_owner/IPetOwner.cairo [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    ) -> u256;
+    fn is_pet_owner(self: @TContractState, id: u256, pet_owner: ContractAddress) -> bool;
+    fn update_owner_profile(
+        ref self: TContractState,
+        id: u256,
+        name: ByteArray,
+        email: ByteArray,
+        emergency_contact: ByteArray,
+    ) -> bool;
+    fn return_pet_owner_info(ref self: TContractState, pet_owner_addr: ContractAddress) -> PetOwner;
```
Comment (manlikeHB) [inline src/components/pet_owner/IPetOwner.cairo:17]: rename to `get_pet_owner`

File: src/components/pet_owner/petowner_component.cairo [hunk 4]
```diff
@@ -0,0 +1,130 @@
+use petchain::components::pet_owner::IPetOwner::{IPetOwner};
+#[starknet::component]
+pub mod PetOwner_component {
```
Comment (manlikeHB) [inline src/components/pet_owner/petowner_component.cairo:3]:
```suggestion
pub mod PetOwnerComponent {
```

File: src/components/pet_owner/petowner_component.cairo [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    #[derive(Drop, starknet::Event)]
+    struct PetOwnerUpdated {
+        #[key]
+        owner_address: ContractAddress,
+        owner_id: u256,
+    }
+
+    #[embeddable_as(PetOwnerComponent)]
```
Comment (manlikeHB) [inline src/components/pet_owner/petowner_component.cairo:37]:
```suggestion
    #[embeddable_as(PetOwnerImpl)]
```

**Review by manlikeHB — CHANGES_REQUESTED** [review]

**Review by manlikeHB — CHANGES_REQUESTED** [review]
Amazing work! last change and we're good.

**Review by manlikeHB — APPROVED** [review]
Thank you @aji70, LGTM!!!
